radio/src/gps.cpp

```diff
@@ -207,12 +228,14 @@ bool gpsNewFrameNMEA(char c)
           switch (gps_frame) {
             case FRAME_GGA:
               frameOK = 1;
+              gpsData.fix = gps_Msg.fix;
+              gpsData.numSat = gps_Msg.numSat;
               if (gps_Msg.fix) {
-                gpsData.fix = 1;
+                __disable_irq();    // do the atomic update of lat/lon
```

Comment:
I think we will need another thread (at least on Horus) soon:
 * the new thread will have priority lower than mixers thread, but higher than main thread
 * new thread would process all periodic things like: internal GPS, gyro/accc sensor, Bluetooth, etc. 

Basically the stuff that is now handled in the main thread, but should really be handled more often and more independently of the Lua/gui cpu usage and relatively high loop times of the main thread.